libdevice/cmath_wrapper.cpp

```diff
@@ -17,13 +17,27 @@ float logf(float x) { return __devicelib_logf(x); }
 
 DEVICE_EXTERN_C
 float expf(float x) { return __devicelib_expf(x); }
-
+// On Windows, the math.h includes wrapper definition for functions:
+// frexpf, ldexpf, hypotf, so we can't provide these 3 functions in
+// device libraries which may lead to multiple definition error.
+// The "hypotf" on Windows will call an internal function "_hypotf"
+// and frexpf, ldexpf will call corresponding double version:
+// frexp and ldexp. frexpf and ldexpf can only be used on platforms
+// with fp64 support currently.
+#ifndef _WIN32
 DEVICE_EXTERN_C
 float frexpf(float x, int *exp) { return __devicelib_frexpf(x, exp); }
 
 DEVICE_EXTERN_C
 float ldexpf(float x, int exp) { return __devicelib_ldexpf(x, exp); }
 
+DEVICE_EXTERN_C
+float hypotf(float x, float y) { return __devicelib_hypotf(x, y); }
+#else
+DEVICE_EXTERN_C
+float _hypotf(float x, float y) { return __devicelib_hypotf(x, y); }
+#endif
+
```

Comment:
I think the order of declarations might be important here. 
Are they in sorted order?
If so I suggest keep `hypotf` declaration location.

Reply:
Hi, Alexey.
I didn't get your point here. What is the "order of declarations" you mentioned?

Reply:
FWIW, it seems to me that function definitions are sorted in lexicographical order. This change brakes this order and I'm not sure why.

Reply:
@jinge90, can you please try to keep the original order of definitions?

Reply:
Hi, @vzakhari .
It seems that the functions in cmath_wrapper/complex_wrapper are not sorted in lexicographical order. For example, the first 3 functions in cmath_wrapper is:
scalbnf,
logf,
expf.
Do I need to sort all functions in cmath/complex_wrapper in lexicographical order?

Reply:
Keeping the original order for hypotf is done.